File: S2M1/src/Bicicleta.java
public abstract class Bicicleta implements Runnable{
        protected int dorsal;
        protected boolean acabado;
        
        public Bicicleta(int dorsal, boolean tengoAcabar){
                this.dorsal = dorsal;
                acabado=tengoAcabar;
        }
	abstract void acabarCarrera();
}

File: S2M1/Ruby/bicicleta.rb
class Bicicleta
    attr_accessor :dorsal, :acabado
    
    def initialize(dorsal, tengoAcabar)
        @dorsal = dorsal
        @acabado = tengoAcabar
    end

    def acabar_carrera
        @acabado = true
    end

    def set_dorsal(val)
        @dorsal = val # `self` is an instance of Foo, it has `bar=` method
        @dorsal
    end
end